subprojects/core-api/src/main/java/org/gradle/api/file/Directory.java

```diff
@@ -79,4 +79,15 @@
      * @return The file.
      */
     Provider<RegularFile> file(Provider<? extends CharSequence> path);
+
+    /**
+     * Returns a {@link FileCollection} containing the given files, whose locations are the given paths resolved relative to this directory.
```

Comment:
Should add a link to Project#files(), which defines the different kinds of paths that can be used.